### PR TITLE
[RSDK-7154] update analog reader interface to include range/accuracy

### PR DESCRIPTION
**Pull request by penguinland**
Surprisingly, the compiled protobufs in this repo already have the new data, and all I needed to do was start using it!

When reading an analog pin on a board, you now get the raw value, the minimum/maximum possible value the reader can output, and the minimum change the reader can notice.

### CODE REVIEW

File: src/viam/components/board/board.py [hunk 1]
```diff
@@ -38,11 +38,23 @@ class Analog:
         name: str
         """The name of the analog pin"""
 
+        class Value:
```
Comment (njooma) [inline src/viam/components/board/board.py:41]: You might want to make this a `@dataclass` to automatically generate an initializer

File: src/viam/components/board/board.py [hunk 2]
```diff
@@ -38,11 +38,23 @@ class Analog:
...
+            """
+            Data obtained from reading an analog pin. The value is the raw bits obtained, while the
+            range indicates indicates the what the extreme values indicate. The step_size is the
+            precision per bit of the reading. The value read, when interpreted in
+            volts/degrees/etc., is (min_range + value * step_size)
+            """
+            value: int
+            min_range: float
+            max_range: float
+            step_size: float
```
Comment (njooma) [inline src/viam/components/board/board.py:51]: Please put the individual attribute docs as docstrings. Then, the class docstring can be how to use this class

```python3
value: int
"""The raw bits obtained"""

min_range: float
"""The minimum of the range"""

max_range: float
"""..."""

step_size: float
"""Precision per bit of the reading"""
```

File: src/viam/components/board/client.py [hunk 3]
```diff
@@ -48,12 +48,17 @@ async def read(
...
         if extra is None:
             extra = {}
         request = ReadAnalogReaderRequest(board_name=self.board.name, analog_reader_name=self.name, extra=dict_to_struct(extra))
         response: ReadAnalogReaderResponse = await self.board.client.ReadAnalogReader(request, timeout=timeout)
-        return response.value
+        result = self.Value()
+        result.value = response.value
+        result.min_range = response.min_range
+        result.max_range = response.max_range
+        result.step_size = response.step_size
```
Comment (njooma) [inline src/viam/components/board/client.py:60]: Once you make `Value` a `@dataclass`, you'll be able to move this into the constructor

**Review by njooma — CHANGES_REQUESTED** [review]

File: src/viam/components/board/board.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (njooma) [reply src/viam/components/board/board.py:41]: I think this is fine. Value is a bit overloaded (particularly with protobuf.Value), but it being namespaced inside of Analog is perfect

File: src/viam/components/board/board.py [hunk 4]
```diff
@@ -38,11 +39,28 @@ class Analog:
         name: str
         """The name of the analog pin"""
 
+        @dataclass
+        class Value:
```
Comment (njooma) [inline src/viam/components/board/board.py:43]: You could probably make this a `TypeAlias` if all the attributes are the same

```
if sys.version_info >= (3, 10):
    from typing import TypeAlias
else:
    from typing_extensions import TypeAlias

class Board:
    class Analog:
        Value: "TypeAlias" = ReadAnalogReaderResponse
```

Similar to how it's done in [Vision](https://github.com/viamrobotics/viam-python-sdk/pull/617/files#diff-a7311669680bf0703ece9bd2c4887bb9bf823cd40ffd8c4154d56c4f38733e15R58)

**Review by njooma — COMMENTED** [review]
Comment, open for discussion

File: src/viam/components/board/board.py (hunk 4 above)
Comment (njooma) [reply src/viam/components/board/board.py:43]: This way, we don't have to keep `Value` in sync with the response type should any fields be added/removed

**Review by stuqdog — APPROVED** [review]
lgtm, modulo @njooma's `TypeAlias` suggestion (which I agree with).

File: src/viam/components/board/board.py (hunk 4 above)
Comment (penguinland) [reply src/viam/components/board/board.py:43]: Oh dang, that's _super_ clean! I had no idea that was a thing. Thanks!

File: src/viam/components/board/client.py [hunk 5]
```diff
@@ -48,12 +48,14 @@ async def read(
...
-    ) -> int:
+    ) -> Board.Analog.Value:
         if extra is None:
             extra = {}
         request = ReadAnalogReaderRequest(board_name=self.board.name, analog_reader_name=self.name, extra=dict_to_struct(extra))
         response: ReadAnalogReaderResponse = await self.board.client.ReadAnalogReader(request, timeout=timeout)
-        return response.value
+        result = self.Value(value=response.value, min_range=response.min_range,
+                            max_range=response.max_range, step_size=response.step_size)
+        return result
```
Comment (njooma) [inline src/viam/components/board/client.py:58]: you should just be able to do `return response` now

**Review by njooma — CHANGES_REQUESTED** [review]

File: src/viam/components/board/client.py (hunk 5 above)
Comment (penguinland) [reply src/viam/components/board/client.py]: Good idea! Take one more look.

File: src/viam/components/board/service.py [hunk 6]
```diff
@@ -139,7 +139,8 @@ async def ReadAnalogReader(self, stream: Stream[ReadAnalogReaderRequest, ReadAna
             raise e.grpc_error
         timeout = stream.deadline.time_remaining() if stream.deadline else None
         value = await analog_reader.read(extra=struct_to_dict(request.extra), timeout=timeout, metadata=stream.metadata)
-        response = ReadAnalogReaderResponse(value=value)
+        response = ReadAnalogReaderResponse(value=value.value, min_range=value.min_range,
+                                            max_range=value.max_range, step_size=value.step_size)
         await stream.send_message(response)
```
Comment (njooma) [inline src/viam/components/board/service.py:144]: You should be able to just do `stream.send_message(value)` now that `value` is of type `ReadAnalogReaderResponse`

**Review by njooma — CHANGES_REQUESTED** [review]

File: src/viam/components/board/service.py (hunk 6 above)
Comment (penguinland) [reply src/viam/components/board/service.py]: Sorry to make you catch all these things that I should have caught myself. I appreciate it; I think this version of the code looks way better than the original!

**Review by njooma — APPROVED** [review]
nice

**Pull request merged**